vm/devices/storage/disk_crypt_resources/src/lib.rs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Resources for the encrypted disk device.
+
+#![warn(missing_docs)]
+
+use mesh::MeshPayload;
+use vm_resource::kind::DiskHandleKind;
+use vm_resource::Resource;
+use vm_resource::ResourceId;
+
+/// A handle to an encrypted disk.
+#[derive(MeshPayload)]
+pub struct DiskCryptHandle {
+    /// The inner disk.
+    pub inner: Resource<DiskHandleKind>,
+    /// The cipher to use for encryption.
+    pub cipher: Cipher,
+    /// The key. This must be appropriately sized for the cipher.
```

Comment:
if there's coupling between the key and the cipher, shouldn't the key be part of the Cipher enum?

Reply:
Arguably. But I suspect we'll end up moving the key into its own kind of "secret" Resource or something, so that the key can be provided through some separate mechanism.

Reply:
You do point to a pet peeve of mine--crypto libraries are notorious for overloading the meaning of things so that every algorithm fits into a single interface of "key + IV + plaintext". And then users screw things up, because the IV has different meanings and requirements for each algorithm.

Reply:
Sure, I wouldn't block the PR on this, but if there's a easy on-ramp to catching those sorts of classic blunders at compile time, that'd be cool